Sources/SKLogging/NonDarwinLogging.swift

```diff
@@ -12,6 +12,40 @@
 
 import SwiftExtensions
 
+#if SOURCEKITLSP_XCTFAIL_ON_FAULT
+import XCTest
+
+private let allowFault = AtomicBool(initialValue: false)
```

Comment:
Do we need to worry about race conditions if e.g one thread enters `withLoggingFaultsAllowed`, and another thread then leaves `withLoggingFaultsAllowed`, leaving this set to `false` for the first thread?